tests/testthat/test-pivot-wide.R

```diff
@@ -885,63 +939,3 @@ test_that("`id_cols` compat behavior doesn't trigger if named `...` are supplied
     pivot_wider(df, ids = id)
   })
 })
-
-# Tests for issue #1609 / #1482 -----------------------------------------------
-
-test_that("doesn't crash when `id_cols` selects column removed by `names_from` (#1609)", {
-  # Original issue scenario
-  x <- c(1, 2, 12, 31, 123, 2341)
-  df <- data.frame(x = x)
-
-  # This should produce a proper tidyselect error, not an internal error
-  expect_snapshot(error = TRUE, {
-    df %>%
-      dplyr::mutate(y = stringr::str_split(x, "")) %>%
-      unnest(cols = y) %>%
-      pivot_wider(
-        id_cols = x,
-        values_from = y,
-        names_from = x
-      )
-  })
```

Comment:
Simplified this pretty dramatically to a minimal reprex with meaningful comments